src/azure-cli/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -1317,6 +1366,11 @@ def _update_event_subscription_internal(  # pylint: disable=too-many-locals,too-
     if included_event_types is not None:
         event_subscription_filter.included_event_types = included_event_types
 
+    event_subscription_filter.enable_advanced_filtering_on_arrays = enable_advanced_filtering_on_arrays
```

Comment:
Should we update `event_subscription_filter.enable_advanced_filtering_on_arrays` only when `enable_advanced_filtering_on_arrays` is not `None`.?

If it used to have a value and customers didn't specify any value by `--enable-advanced-filtering-on-arrays`, this might update the original value to `None`, right?

Reply:
yes you are right. I did this before to debug. thanks !